Cython/Compiler/Nodes.py

```diff
@@ -6285,6 +6287,32 @@ def analyse_types(self, env, use_temp=0):
 
         self.rhs = self.rhs.analyse_types(env)
 
+        if self.lhs.is_name and self.lhs.entry.type.is_const:
+            if self.lhs.entry.type.cv_base_type.is_ptr:
+                # Currently, `cdef TYPE *const ...` is not supported
+                error(self.pos, f"Assignment to const '{self.lhs.name}'")
+
+            elif env.is_module_scope and self.lhs.entry.init is None and (
+                    self.rhs.has_constant_result() or self.rhs.is_literal or self.rhs.is_name and self.rhs.type.is_const):
+                # const variable can be initialised if:
+                # * we are in module scope
+                # * self.lhs was not assigned before
+                # * self.rhs is literal, const expression or const variable
+                if self.rhs.is_literal:
+                    # case of assignment a value: `cdef const float my_var = 2.0`
+                    self.lhs.entry.init = self.rhs.constant_result
```

Comment:
You first test for `rhs.has_constant_result() or rhs.is_literal`, thus expecting literals to not (necessarily) have a constant result, and then you use `if rhs.is_literal: init = rhs.constant_result`. That seems contradictory.